src/features/unified-share-modal/__tests__/UnifiedShareModal-test.js

```diff
@@ -738,4 +620,207 @@ describe('features/unified-share-modal/UnifiedShareModal', () => {
             expect(wrapper.instance().shouldAutoFocusSharedLink()).toBe(true);
         });
     });
+
+    describe('onPillCreate()', () => {
+        const email = 'dev@box.com';
+        const displayText = email;
+        const name = 'dev';
+        const text = name;
+        const id = 123;
+        const type = 'user';
+        const isExternalUser = true;
+        const value = email;
+        const inviteCollabsContacts = [
+            {
+                displayText: email,
+                value: email,
+            },
+        ];
+        const emailSharedLinkContacts = [
+            {
+                displayText: email,
+                value: email,
+            },
+        ];
+
+        test('should not retrieve emails if it is a select dropdown case where there is only one pill with id', async () => {
+            const pills = [{ id: 123 }];
+            const getContactsByEmail = jest.fn();
+            const wrapper = getWrapper({ getContactsByEmail });
+            await wrapper.instance().onPillCreate(pills);
+            expect(getContactsByEmail).not.toHaveBeenCalled();
+        });
+
+        test('should retrieve emails and set the state for invite collabs contacts', async () => {
```

Comment:
Should extend or create a new test that passes a mix of selectType and contactType pills (as per the change I mentioned above)

Reply:
done